docs/boundary-services-a-to-z.adoc

```diff
@@ -0,0 +1,183 @@
+= Boundary Services A-Z
+
+This document describes how to run a RAP environment with boundary services.
+It's a quick rundown and assumes knowledge of the basic setup described in the
+Running Omicron (Non-Simulated) document.
+
+Since we're using virtual network hardware, the OPTE hack is no longer needed.
+The hack will automatically be disabled during the `create_virtual_hardware.sh`
+script.
+
+== 0. Install softnpu ASIC emulator machinery
+
+----
+./tools/install_softnpu_machinery.sh
+----
+
+== 1. Setup virtual hardware
+
+----
+PHYSICAL_LINK=<wan interface> pfexec ./tools/create_virtual_hardware.sh
+----
+Note that the `PHYSICAL_LINK` environment variable is optional. If not supplied,
+the first link in `dladm show-phys` will be used.
+
+The virtual hardware is a bit different than what's currently being used. What
+we'll eventually wind up with looks like this.
+
+image::plumbing.png[]
+
+The `softnpu` zone will be configured and launched during the `create_virtual_hardware.sh`
+script.
+
+== 2. Build and install the control plane.
+
+----
+./tools/create_self_signed_cert.sh
+
+cargo build --release --bin omicron-package
+./target/release/omicron-package -t switch_variant=softnpu package
```

Comment:
Looks like if we're overriding the default we now need to explicitly set the `image_type` option.

```suggestion
./target/release/omicron-package -t 'image_type=standard switch_variant=softnpu' package
```

Reply:
Looks like the latest package magic requires
```
./target/release/omicron-package -t <target_name> target create -i standard -m non-gimlet -s softnpu
```

I've updated the `a-to-z` doc accordingly.